extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java

```diff
@@ -55,72 +56,73 @@ public void startVaultContainers(BuildProducer<DevServicesConfigResultBuildItem>
             }
-            closeables = null;
             capturedDevServicesConfiguration = null;
         }
 
         capturedDevServicesConfiguration = currentDevServicesConfiguration;
 
-        StartResult startResult;
+        VaultInstance vaultInstance;
 
         StartupLogCompressor compressor = new StartupLogCompressor(
                 (launchMode.isTest() ? "(test) " : "") + "Vault Dev Services Starting:", consoleInstalledBuildItem,
                 loggingSetupBuildItem);
         try {
-            startResult = startContainer(currentDevServicesConfiguration, devServicesConfig.timeout);
+            vaultInstance = startContainer(currentDevServicesConfiguration, launchMode, devServicesConfig.timeout);
+            if (vaultInstance != null) {
+                closeable = vaultInstance.getCloseable();
+
+                devConfig.produce(new DevServicesConfigResultBuildItem(URL_CONFIG_KEY, vaultInstance.url));
+                devConfig.produce(new DevServicesConfigResultBuildItem(CLIENT_TOKEN_CONFIG_KEY, vaultInstance.clientToken));
+
+                if (vaultInstance.isOwner()) {
+                    log.info("Dev Services for Vault started.");
+                    log.infof("Other Quarkus applications in dev mode will find the "
+                            + "instance automatically. For Quarkus applications in production mode, you can connect to"
+                            + " this by starting your application with -D%s=%s -D%s=%s",
+                            URL_CONFIG_KEY, vaultInstance.url, CLIENT_TOKEN_CONFIG_KEY, vaultInstance.clientToken);
+                }
+            }
             compressor.close();
         } catch (Throwable t) {
             compressor.closeAndDumpCaptured();
             throw new RuntimeException(t);
         }
-        if (startResult == null) {
-            return;
-        }
-        Map<String, String> connectionProperties = new HashMap<>();
-        connectionProperties.put(URL_CONFIG_KEY, startResult.url);
-        connectionProperties.put(CLIENT_TOKEN_CONFIG_KEY, startResult.clientToken);
-
-        closeables = Collections.singletonList(startResult.closeable);
 
         if (first) {
             first = false;
             Runnable closeTask = new Runnable() {
                 @Override
                 public void run() {
-                    if (closeables != null) {
-                        for (Closeable closeable : closeables) {
-                            try {
-                                closeable.close();
-                            } catch (Throwable t) {
-                                log.error("Failed to stop Vault container", t);
-                            }
+                    if (closeable != null) {
+                        try {
+                            closeable.close();
+                        } catch (Throwable t) {
+                            log.error("Failed to stop Vault container", t);
```

Comment:
should probably null out `closeable` in finally for consistency with the first occurence. in this case you are doing it anyway, after catching `Throwable`, so you are ok.

Reply:
Moved inside if check for better readability.